src/Features/Core/Portable/GenerateConstructorFromMembers/AbstractGenerateConstructorFromMembersCodeRefactoringProvider.cs

```diff
@@ -47,60 +49,149 @@ protected AbstractGenerateConstructorFromMembersCodeRefactoringProvider() : this
         /// <summary>
         /// For testing purposes only.
         /// </summary>
-        protected AbstractGenerateConstructorFromMembersCodeRefactoringProvider(IPickMembersService pickMembersService_forTesting)
+        protected AbstractGenerateConstructorFromMembersCodeRefactoringProvider(IPickMembersService? pickMembersService_forTesting)
             => _pickMembersService_forTesting = pickMembersService_forTesting;
 
         protected abstract bool ContainingTypesOrSelfHasUnsafeKeyword(INamedTypeSymbol containingType);
         protected abstract string ToDisplayString(IParameterSymbol parameter, SymbolDisplayFormat format);
         protected abstract bool PrefersThrowExpression(DocumentOptionSet options);
 
-        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        public override Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            return ComputeRefactoringsAsync(context.Document, context.Span,
+                (action, applicableToSpan) => context.RegisterRefactoring(action, applicableToSpan),
+                (actions) => context.RegisterRefactorings(actions), context.CancellationToken);
```

Comment:
it's an interestnig pattern.  will be worth seeing if we can pull it out to a common place (maybe a base class) in the future so that as we do more intents we can not repeat this everywhere.